lib/kintsugi/cli.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2021 Lightricks. All rights reserved.
+# Created by Ben Yohay.
+# frozen_string_literal: true
+
+require "optparse"
+
+require_relative "version"
+
+module Kintsugi
+  module CLI
+    Command = Struct.new(:option_parser, :action, :description, keyword_init: true)
+
```

Comment:
not sure if you can make stuff private, but in any case put some documentation in this file as currently it has no context.